iothub/device/tests/Authentication/EdgeDeviceOnBehalfOfTests.cs

```diff
@@ -0,0 +1,134 @@
+﻿using Microsoft.Azure.Devices.Client;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Text;
+using System.Threading.Tasks;
+using System;
+using FluentAssertions;
+using Microsoft.Azure.Devices.Client.Authentication;
+
+namespace Microsoft.Azure.Devices.Client.Tests.OnBehalfOf
+{
+    [TestClass]
+    public class EdgeDeviceOnBehalfOfTests
+    {
+        private static string _testKey => Convert.ToBase64String(Encoding.UTF8.GetBytes(Guid.Empty.ToString("N")));
+
+        [TestMethod]
+        public async Task ConnectDeviceOnBehalfOf()
+        {
+            string leafDeviceId = "test-leaf-device";
+            string edgeDeviceId = "test-edge-device";
+
+            var edgeHubCs = new IotHubConnectionString("e4k-hub.azure-devices.net", null, edgeDeviceId, "$edgeHub", null, _testKey, null);
+
+            IAuthenticationMethod leafAuth = new ClientAuthenticationForEdgeHubOnBehalfOf(
+                edgeHubCs.SharedAccessKey!,
+                edgeHubCs.DeviceId!,
+                leafDeviceId,
+                null,
+                TimeSpan.FromMinutes(10),
+                5);
+
+            IotHubDeviceClient leafClient = new(edgeHubCs.IotHubHostName, leafAuth,
+                new IotHubClientOptions(
+                    new IotHubClientAmqpSettings
+                    {
+                        ConnectionPoolSettings = new AmqpConnectionPoolSettings()
+                        {
+                            UsePooling = true,
+                            MaxPoolSize = 10
+                        }
+                    }));
+
+            await leafClient.OpenAsync();
+            long tick = Environment.TickCount;
+            await leafClient.UpdateReportedPropertiesAsync(new ReportedProperties { ["tick"] = tick });
+            var twin = await leafClient.GetTwinPropertiesAsync();
+            twin.Should().NotBeNull();
+            twin.Reported.Should().NotBeNull();
+            twin.Reported["tick"].Should().NotBeNull();
+            tick.Should().Be((long)twin.Reported["tick"]);
+            await leafClient.CloseAsync();
+            await leafClient.DisposeAsync();
+        }
+
+        [TestMethod]
+        public async Task ConnectLeafModuleOnBehalfOf()
```

Comment:
Would it be better to consolidate two tests to call a single test function with different inputs for easier maintenance given both tests are similar?